packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTree.tsx

```diff
@@ -205,6 +206,7 @@ function useTreeState({
           : selectionPredicateRef.current(node.key, ModelsVisibilityHandler.getNodeType(node)),
       [],
     ),
+    eventHandler: useCallback((props: VisibilityTreeEventHandlerParams) => new ModelsTreeEventHandler(props), []),
```

Comment:
Maybe better to define this factory outside of render function? Wouldn't need `useCallback` then.

Reply:
Not having it wrapped causes infinite re-renders. `selectionPredicate` parameter is passed in the same way. Also, it makes sense to wrap the event handler in a` useCallback` with no dependencies, since this passed `eventHandler` factory is specific for models tree and is not subject to change.

Reply:
Grigas clarified his suggestion in private, applied it.